Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -36,11 +58,14 @@ void InfoComponentVisitor::registerComponentAssembly(
     child->registerContents(*this);
   }
   const size_t detectorStop = m_assemblySortedDetectorIndices.size();
-
   m_ranges.emplace_back(std::make_pair(detectorStart, detectorStop));
 
   // For any non-detector we extend the m_componetIds from the back
   m_componentIds.emplace_back(assembly.getComponentID());
+
+  m_positions->emplace_back(Kernel::toVector3d(assembly.getPos()));
+  m_rotations->emplace_back(Kernel::toQuaterniond(assembly.getRotation()));
+  clearPositionAndRotationsParameters(m_pmap, assembly);
```

Comment:
Yes. I'll add a test for this in any case. 

As a side point, you'll notice that in visiting detectors, no positions/rotation are stored. One improvement that I'd like to make (in a future PR) is to have the visitor create the `DetectorInfo` as well as the `ComponentInfo`. There are several potential benefits to that.

1. It would consolidate the place that all relevant `IComponent` subtypes are iterated through and used to create info layers. 
2. We visit the `IDetectors` already anyway, so the `DetectorInfo` generation should slot in nicely.
3. `ComponentInfo` depends on `DetectorInfo` now, but I have to expose lots of the getters of the `InfoComponentVisitor` just so that I can bind the two together. Doing this internally would simplify that API quite a lot.